File: sort/Bucketsort/Java/BucketSort.java
package Java;

public class BucketSort {

    private BucketSort() { }

    private static int[] bucketSort(int maxInt, int arrayToSort[]){
        int buckets[] = new int[maxInt];

        for (int i = 0; i < arrayToSort.length-1; i++) {
            buckets[arrayToSort[i]] = arrayToSort[i];
        }

        return buckets;
    }

    private static void printArray(int array[]){
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
    }

    public static void main(String args[]) {
        int arr[] = {64, 34, 25, 12, 22, 11, 90};
        int maxInt = 100;

        int sortedArr[] = bucketSort(maxInt, arr[]);

        System.out.println("Sorted array: ");
        printArray(sortedArr);
    }
}

File: sort/Bucketsort/Python/bucketSort.py
def bucketsort(maxint, arraytosort):
    buckets = [0] * maxint
    i = 0
    while i < len(arraytosort):
        buckets[arraytosort[i]] = arraytosort[i]
        i += 1
    return buckets


arraytosort = [64, 34, 25, 12, 22, 11, 90]
maxint = 100

sortedarray = bucketsort(maxint, arraytosort)

print ("Sorted array is:")
print(sortedarray, end='')